### PR TITLE
refactor: overhaul and cleanup rule exclusions for settings

### PR DESCRIPTION
**Pull request by EsadCetiner**
Over the past year the Nextcloud plugin has grown dramatically in size with improved coverage and support for various features within Nextcloud, this has come at the expense of code readability and maintainability. Rule IDs have grown from 53 to 185 and the lines of code from 780 to 2756.

This PR is focused on overhauling personal/administration settings and has:

- Reduced the number of rules down to 181 (from 185)
- Rule exclusions have been reorganized to make it easier to find/modify/add rule exclusions relating to settings
- Comment clarity has been improved
- 6 Additional tests have been added

The Nextcloud plugin will likely remain large since Nextcloud incredibly feature rich, but it's a start.

### CODE REVIEW

File: plugins/nextcloud-rule-exclusions-before.conf [hunk 1]
```diff
@@ -1570,144 +1610,186 @@ SecRule REQUEST_FILENAME "@contains /settings/api/admin/twofactorauth" \
...
+    ctl:ruleRemoveTargetById=931130;ARGS:json.value"
 
-# When disabling settings under Settings --> Availability
-SecRule REQUEST_FILENAME "@rx /ocs/v[0-9]\.php/apps/provisioning_api/api/v[0-9]/config/users/dav/user_status_automation$" \
-    "id:9508619,\
+#
+# [ Administration settings ]
+#
+
+# Personal Settings --> Privacy --> Who has access to your data?
```
Comment (theseion) [inline plugins/nextcloud-rule-exclusions-before.conf:1686]: This one should go under personal settings, right?

File: plugins/nextcloud-rule-exclusions-before.conf [hunk 2]
```diff
@@ -1570,144 +1610,186 @@ SecRule REQUEST_FILENAME "@contains /settings/api/admin/twofactorauth" \
...
-    ctl:ruleRemoveTargetByTag=OWASP_CRS;ARGS:newpassword,\
-    ctl:ruleRemoveTargetByTag=OWASP_CRS;ARGS:json.oldpassword,\
-    ctl:ruleRemoveTargetByTag=OWASP_CRS;ARGS:json.newpassword"
+    setvar:'tx.allowed_methods=%{tx.allowed_methods} DELETE'"
 
-# Validating password against password policy
-SecRule REQUEST_FILENAME "@rx /ocs/v[0-9]\.php/apps/password_policy/api/v[0-9]/validate$" \
-    "id:9508626,\
+# Administration Settings --> Theming --> Advanced Options
+# Adding links foir legal notices and privacy policies
```
Comment (theseion) [inline plugins/nextcloud-rule-exclusions-before.conf:1776]:
```suggestion
# Adding links for legal notices and privacy policies
```

**Review by theseion — CHANGES_REQUESTED** [review]
You didn't change any of the existing rules (except for dropping the `^` anchor in one place), right?

File: plugins/nextcloud-rule-exclusions-before.conf (hunk 1 above)
Comment (EsadCetiner) [reply plugins/nextcloud-rule-exclusions-before.conf:1686]: This is shown in personal settings within Nextcloud's UI, but only admins are able to add/remove users to "who can see your data"

**Comment by EsadCetiner** [comment]
@theseion I didn't really touch any of the functionality of the rules. they largely work the same. It's just a reorganization/cleanup and simplifying

**Review by theseion — APPROVED** [review]

**Pull request merged**